ethers-providers/src/lib.rs

```diff
@@ -263,6 +266,14 @@ pub trait Middleware: Sync + Send + Debug {
         self.inner().lookup_address(address).await.map_err(FromErr::from)
     }
 
+    async fn resolve_avatar(&self, ens_name: &str) -> Result<Url, Self::Error> {
+        self.inner().resolve_avatar(ens_name).await.map_err(FromErr::from)
+    }
+
+    async fn resolve_field(&self, ens_name: &str, field: &str) -> Result<String, Self::Error> {
```

Comment:
please document these trait functions, or link to the matching functions, that are document.